block-node/publisher/src/test/java/org/hiero/block/node/publisher/BlockStreamProducerSessionTest.java

```diff
@@ -0,0 +1,382 @@
+// SPDX-License-Identifier: Apache-2.0
+package org.hiero.block.node.publisher;
+
+import static org.hiero.block.node.spi.BlockNodePlugin.UNKNOWN_BLOCK_NUMBER;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.hedera.pbj.runtime.grpc.Pipeline;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.metrics.api.Counter;
+import java.util.List;
+import java.util.concurrent.Flow;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Consumer;
+import org.hiero.block.node.spi.blockmessaging.BlockItems;
+import org.hiero.hapi.block.node.BlockItemUnparsed;
+import org.hiero.hapi.block.node.PublishStreamResponse;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/**
+ * Unit tests for {@link BlockStreamProducerSession}.
+ * Tests cover session state management, block processing, and response handling.
+ */
+@ExtendWith(MockitoExtension.class)
+@DisplayName("Block Stream Producer Session Tests")
+public class BlockStreamProducerSessionTest {
+
+    @Mock
+    private UpdateCallback onUpdate;
+
+    @Mock
+    private Consumer<BlockItems> sendToBlockMessaging;
+
+    @Mock
+    private Counter liveBlockItemsReceived;
+
+    private BlockStreamProducerSession session;
+
+    private PublishStreamResponse lastResponse;
+
+    private Bytes BLOCK_HEADER = Bytes.fromHex(
```

Comment:
We should have one way to get sample block data for tests. I added `org.hiero.block.node.app.fixtures.blocks.SimpleTestBlockItemBuilder` that can probably be used here and also @AlfredoG87  is looking at adding a full block of sample data to the fixture.